rct229/rules/section5/section5rule37.py

```diff
@@ -0,0 +1,290 @@
+from rct229.ruleset_functions.get_building_scc_skylight_roof_ratios_dict import (
+    get_building_scc_skylight_roof_ratios_dict,
+)
+from rct229.ruleset_functions.get_opaque_surface_type import OpaqueSurfaceType as OST
+from rct229.ruleset_functions.get_opaque_surface_type import get_opaque_surface_type
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    SurfaceConditioningCategory as SCC,
+)
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    get_surface_conditioning_category_dict,
+)
+from rct229.utils.pint_utils import ZERO
+from rct229.utils.std_comparisons import std_equal
+
+MANUAL_CHECK_MSG = "Manual review is required to verify skylight meets U-factor requirement as per table G3.4."
+MANUAL_CHECK_APPLICABLE = (
+    "The subsurface type is Door, not applicable for the rule-checking"
+)
+DOOR = schema_enums["SubsurfaceClassificationType"].DOOR.name
+
+
+class Section5Rule37(RuleDefinitionListIndexedBase):
+    """Rule 37 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule37, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section5Rule37.BuildingRule(),
+            index_rmr="baseline",
+            id="5-37",
+            description="Skylight U-factors for residential, non-residential and semi-heated spaces in the baseline model must match the appropriate requirements in Table G3.4-1 through G3.4-8.",
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    def create_data(self, context, data=None):
+        rmr_baseline = context.baseline
+        return {"climate_zone": rmr_baseline["weather"]["climate_zone"]}
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule37.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section5Rule37.BuildingRule.BuildingSegmentRule(),
+                index_rmr="baseline",
+                list_path="building_segments[*]",
+                manual_check_required_msg=MANUAL_CHECK_MSG,
+            )
+
+        def create_data(self, context, data=None):
+            building_b = context.baseline
+            climate_zone = data["climate_zone"]
+            # Merge into the existing data dict
+            return {
+                **data,
+                "building_scc_skylight_roof_ratios_dict_b": get_building_scc_skylight_roof_ratios_dict(
+                    climate_zone, building_b
+                ),
+                "surface_conditioning_category_dict_b": get_surface_conditioning_category_dict(
+                    climate_zone, building_b
+                ),
+            }
+
```

Comment:
By my reading of the RDS, all the output should be at the Building level.

I am very confused about the idea of looping over building segments. If anything, it seems that we would go straight to surfaces.

Reply:
Its good that you have checked the RDS... I forgot the function returns building level SRR and the rule RDS has the same mistake. I will update both.